main.tf

```diff
@@ -122,80 +114,61 @@ module "eks" {
   }
   iam_role_additional_policies = {
     AmazonEBSCSIDriverPolicy = "arn:aws:iam::aws:policy/service-role/AmazonEBSCSIDriverPolicy",
-    SecretsManagerReadWrite = "arn:aws:iam::aws:policy/SecretsManagerReadWrite"
+    SecretsManagerReadWrite  = "arn:aws:iam::aws:policy/SecretsManagerReadWrite"
   }
-  # aws-auth configmap
-  manage_aws_auth_configmap = true
 
-  aws_auth_roles = [
-    {
-      rolearn  = aws_iam_role.admin_role.arn
-      username = "admin"
-      groups   = ["system:masters"]
-    },
-  ]
+  # Cluster access entry
+  # To add the current caller identity as an administrator
+  enable_cluster_creator_admin_permissions = true
+  authentication_mode                      = "API"
+
+
+  access_entries = {
+    # One access entry with a policy associated
+    eks_admin_role = {
+      kubernetes_groups = []
+      principal_arn     = "arn:aws:iam::766808016710:role/eks_admin_role"
+
+      policy_associations = {
+        eks_admin_role = {
+          policy_arn = "arn:aws:eks::aws:cluster-access-policy/AmazonEKSAdminPolicy"
+          access_scope = {
+            type = "cluster"
+          }
+        }
+      }
+    }
+  }
   tags = var.tags
 }
 
+module "ocean-controller" {
```

Comment:
@thomasyu888 Something I could use some help on. I wasn't clear based on their published docs which modules I was really supposed to use here. Specifically the issue I am seeing in the Spot console is:


```
Different controller versions detected. Please remove the old version controller. Old version controller pod(s): [spotinst-kubernetes-cluster-controller-685b9885bb-7nll9], New version controller pod(s): [ocean-controller-ocean-kubernetes-controller-77b5cb4c7-6dt4g, ocean-controller-ocean-kubernetes-controller-77b5cb4c7-wbslc] 
```

![image](https://github.com/thomasyu888/eks-stack/assets/17128019/4b8fd8a5-3f73-47c0-a164-3885ee65b85c)